src/ln/channelmonitor.rs

```diff
@@ -677,9 +882,29 @@ impl ChannelMonitor {
 				} else {
 					writer.write_all(&[0; 1])?;
 				}
-
+				match funding_info  {
+					&Some((ref outpoint, ref script)) => {
+						writer.write_all(&outpoint.txid[..])?;
+						writer.write_all(&byte_utils::be16_to_array(outpoint.index))?;
+						script.write(writer)?;
+					},
+					&None => {
+						// We haven't even been initialized...not sure why anyone is serializing us, but
+						// not much to give them.
+						return Ok(());
+					},
+				}
+				match short_channel_id {
+					&Some(short_channel_id) => {
+						writer.write_all(&[1; 1])?;
+						writer.write_all(&byte_utils::be64_to_array(short_channel_id))?;
+					}
+					&None => {
+						writer.write_all(&[0; 1])?;
```

Comment:
nit: just write out 8 bytes of 0s?

Reply:
No 8 bytes of 0s fails serialization test, because you get Some(&[0;8]) instead of None at deserialization!